src/main/io/osd.c

```diff
@@ -479,6 +480,42 @@ static void osdDrawSingleElement(uint8_t item)
         break;
     }
 
+    case OSD_MAIN_BATT_USAGE:
+    {
+        //Set length of indicator bar
+        uint8_t progressSteps = 10;
+
+        //Calculate constrained value
+        float value = constrain(osdConfig()->cap_alarm - getMAhDrawn(), 0, osdConfig()->cap_alarm);
+
+        //Calculate mAh used progress
+        uint8_t mAhUsedProgress = ceil((value / (osdConfig()->cap_alarm / progressSteps)));
+
+        //Create empty battery indicator bar
+        buff[0] = SYM_PB_START;
+        for(uint8_t i = 1; i <= progressSteps; i++)
```

Comment:
You can make this more efficient by doing this after the indicator bar has been filled, and only filling the unused slots.